src/compiler/checker.ts

```diff
@@ -14303,7 +14309,17 @@ namespace ts {
             return true;
         }
 
+        function callLikeExpressionMayHaveTypeArguments(node: CallLikeExpression): node is CallExpression | NewExpression {
+            return node.kind === SyntaxKind.CallExpression || node.kind === SyntaxKind.NewExpression;

```

Comment:
I couldn't find one, but I've added one now. Still want to keep `isCallOrNewExpression` separate from `callLikeExpressionMayHaveTypeArgumets` since those may not always be the same thing (#11947).